src/ceph-volume/ceph_volume/util/system.py

```diff
@@ -33,21 +33,31 @@ def generate_uuid():
 
 def which(executable):
     """find the location of an executable"""
-    locations = (
+    def _get_path(executable, locations):
+        for location in locations:
+            executable_path = os.path.join(location, executable)
+            if os.path.exists(executable_path) and os.path.isfile(executable_path):
+                return executable_path
```

Comment:
Let's either return `None` or an empty string here in case the executable is not found...for explicitness.

Reply:
Done